File: android/app/src/main/java/ai/thetahealth/mirobody/data/auth/AuthDtos.kt
package ai.thetahealth.mirobody.data.auth

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EmailLoginRequest(val email: String)

@Serializable
data class EmailLoginResponse(val email: String? = null)

@Serializable
data class EmailVerifyRequest(val email: String, val code: String)

@Serializable
data class FirebaseVerifyRequest(val token: String)

/**
 * WeChat OpenSDK login. `code` is the OAuth code from a SendAuth.Resp; `flow`
 * tells the backend to exchange it via sns/oauth2/access_token with the mobile
 * app credentials (WECHAT_APP_*).
 */
@Serializable
data class WechatVerifyRequest(val code: String, val flow: String = "app")

@Serializable
data class AuthTokenResponse(
    @SerialName("access_token") val accessToken: String,
    @SerialName("token_type") val tokenType: String = "Bearer",
    @SerialName("expires_in") val expiresIn: Long = 0,
    @SerialName("refresh_token") val refreshToken: String? = null,
    @SerialName("webauthn_registered") val webauthnRegistered: Boolean? = null,
    @SerialName("fallback_token") val fallbackToken: String? = null,
)

File: ios/Mirobody/Data/Auth/AuthDtos.swift
import Foundation

/// Request/response DTOs for the auth endpoints. Mirrors `data/auth/AuthDtos.kt`.

struct EmailLoginRequest: Encodable {
    let email: String
}

struct EmailLoginResponse: Decodable {
    let email: String?
}

struct EmailVerifyRequest: Encodable {
    let email: String
    let code: String
}

struct FirebaseVerifyRequest: Encodable {
    let token: String
}

/// Apple ID token from `ASAuthorizationController`, verified by `/apple/verify`.
struct AppleVerifyRequest: Encodable {
    let idToken: String
    enum CodingKeys: String, CodingKey { case idToken = "id_token" }
}

/// WeChat OpenSDK OAuth code; `flow=app` selects the mobile-app credentials on
/// the backend's `/wechat/verify`.
struct WechatVerifyRequest: Encodable {
    let code: String
    let flow: String
}

struct AuthTokenResponse: Decodable {
    let accessToken: String
    let tokenType: String
    let expiresIn: Int
    let refreshToken: String?
    let webauthnRegistered: Bool?
    let fallbackToken: String?

    enum CodingKeys: String, CodingKey {
        case accessToken = "access_token"
        case tokenType = "token_type"
        case expiresIn = "expires_in"
        case refreshToken = "refresh_token"
        case webauthnRegistered = "webauthn_registered"
        case fallbackToken = "fallback_token"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        accessToken = try c.decode(String.self, forKey: .accessToken)
        tokenType = try c.decodeIfPresent(String.self, forKey: .tokenType) ?? "Bearer"
        expiresIn = try c.decodeIfPresent(Int.self, forKey: .expiresIn) ?? 0
        refreshToken = try c.decodeIfPresent(String.self, forKey: .refreshToken)
        webauthnRegistered = try c.decodeIfPresent(Bool.self, forKey: .webauthnRegistered)
        fallbackToken = try c.decodeIfPresent(String.self, forKey: .fallbackToken)
    }
}
